### PR TITLE
ASM Serverless support for NodeJS and Python

### PR DESCRIPTION
**Pull request by Julio-Guerra**
### What does this PR do?

Start an AWS Lambda Runtime API proxy in the extension to monitor with ASM the AWS Lambda invocation events and results. To do so, invocation event payloads are obtained through Runtime API calls, but invocation responses are rather obtained through the invocation span which contains the data ASM needs (the status code). First versions of this implementation relied on the Runtime API calls in order to obtain the response payload too but it wasn't working with the python tracer which was sending its traces before the call to the Runtime API with the response. Therefore, ASM wasn't unable to send its span tags. The current solution of relying on the span reception as "end of invocation event" is not ideal but works well for our monitoring-only purposes for now.

The ASM Runtime API proxy is enabled when both `DD_SERVERLESS_APPSEC_ENABLED` and `DD_EXPERIMENTAL_ENABLE_PROXY` are enabled. The proxy env var is required in the end because the language is unfortunately unknown when the extension starts and we couldn't rely on `tags.getRuntime()` while the lambda function is being started.

### Motivation

Add Application Security Monitoring for AWS Lambda NodeJS and Python which do not support the Universal Instrumentation API for now.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/serverless/main.go [hunk 1]
```diff
@@ -264,15 +266,18 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 	}()
 
 	// start appsec
-	var httpsecSubProcessor invocationlifecycle.InvocationSubProcessor
+	var (
```
Comment (maxday) [inline cmd/serverless/main.go:269]: could we move all this block in an appsec specific package and call just the entrypoint here?

File: cmd/serverless/main.go [hunk 2]
```diff
@@ -301,12 +306,51 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
...
-	// start the experimental proxy if enabled
-	_ = proxy.Start(
-		"127.0.0.1:9000",
-		"127.0.0.1:9001",
-		serverlessDaemon.InvocationProcessor,
-	)
+	// NodeJS- and Python-specific support for AppSec: monitor the invocation through the runtime API monitoring until
+	// these tracers support the extension's universal instrumentation API
+	// TODO: remove the following block when the python and nodejs tracers support the extension universal instrumentation api
+	if rt := tags.GetRuntime(); appsecInstance != nil && (strings.Contains(rt, "nodejs") || strings.Contains(rt, "python")) {
```
Comment (maxday) [inline cmd/serverless/main.go:312]: same here, could we extract the logic related to appsec in the appsec package and call it from here?

Comment (Julio-Guerra) [reply cmd/serverless/main.go]: Done in [5c782e7](https://github.com/DataDog/datadog-agent/pull/15378/commits/5c782e73f966af69709a01796f6c434fbb51c040)

File: cmd/serverless/main.go (hunk 1 above)
Comment (Julio-Guerra) [reply cmd/serverless/main.go:264]: Done in [5c782e7](https://github.com/DataDog/datadog-agent/pull/15378/commits/5c782e73f966af69709a01796f6c434fbb51c040)

**Comment by Julio-Guerra** [comment]
@maxday, I simplified the source code changes in the main function.
The proxy vs universal instrumentation modes are pretty different so I couldn't strictly unify both.

File: cmd/serverless/main.go [hunk 3]
```diff
@@ -287,22 +288,30 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 	log.Debug("Setting ColdStartSpanCreator on Daemon")
 	serverlessDaemon.SetColdStartSpanCreator(coldStartSpanCreator)
 
+	ta := serverlessDaemon.TraceAgent.Get()
+
```
Comment (maxday) [inline cmd/serverless/main.go:292]: in a follow-up PR maybe it's worth checking ta against not nil?

**Review by maxday — APPROVED** [review]
LGTM! 🚀

**Pull request merged**